lib/matplotlib/tests/test_axes.py

```diff
@@ -602,26 +602,21 @@ def test_fill_units():
     dt = np.arange('2009-04-27', '2009-04-29', dtype='datetime64[D]')
     dtn = mdates.date2num(dt)
 
-    fig = plt.figure()
+    fig, ((ax1, ax2), (ax3, ax4)) = plt.subplots(2, 2)
```

Comment:
I am not a fan of this idiom versus `fig, axs = plt.subplots(2, 2)`.  You lose the ability to iterate if you spell them all out like this.

Reply:
I think which is better depends on what you are doing (a grid of similar Axes or a grid of very different Axes).  

In this case the code was previously using `axN` so this is the minimal touch to move away from `add_subplot(XYZ)`.

Reply:
And I think it also depends on the names you use,

```python
fig, (ax_hist, ax_map) = plt.subplots(2)
```

seems much better that slicing into an array (which is similar to the argument of why `subplot_mosaic` returns a dict).

Reply:
I agree with @tacaswell, especially since this minimizes changes.